pkg/controller/nodes/executor.go

```diff
@@ -403,7 +406,16 @@ func (c *nodeExecutor) handleNode(ctx context.Context, w v1alpha1.ExecutableWork
 		logger.Errorf(ctx, "failed Execute for node. Error: %s", err.Error())
 		return executors.NodeStatusUndefined, err
 	}
-
+	execErr := p.GetErr()
```

Comment:
I think putting this code here will cause double counting. because -for example- we will go here when the node fails, then it moves to failing, then we will go here again after finalize...

Reply:
Also, do we want to mark spot instance disappearing as a system error? maybe a question to @surindersinghp

Reply:
I think yes for now. And as we are able to pinpoint interruptions, we can add one more type (or via label).